### PR TITLE
Add full mathematical formulation of the problem

### LINKED ISSUE #6: Documentation
Within the documentation we can include a section on the mathematical model behind it :)

### PR DESCRIPTION
**Pull request by drvinceknight**
This reference section is not necessarily intended to be very user friendly (as opposed to the tutorial and the howtos etc). That sounds terrible and horribly ivory towerish but what I'm trying to say is that this is EDIT _meant to be_ the precise mathematical formulation, if it can be made clearer it of course should be. However I think it's not necessarily aimed at every potential user of the tool (for example the summation notation $\sum$ (Sigma) might not be as immediately accessible as a python `sum`). This is why it's in the background section.
 
I suggest we close #6 (woops just saw my name for this branch was wrong) and open up specific issues for parts of the documentation we want to see.

### CODE REVIEW

**Comment by meatballs** [comment]
> I suggest we close #6.....

agreed

File: docs/background/mathematical_model.rst [hunk 1]
```diff
@@ -1,8 +1,9 @@
 Mathematical model
 ==================
 
+
 The scheduler work by solving a well understood mathematical problem called an
```
Comment (meatballs) [inline docs/background/mathematical_model.rst]: typo: works

File: docs/background/mathematical_model.rst [hunk 2]
```diff
@@ -55,4 +59,168 @@ be a bit more complicated:
 
 The mathematical representation for these constraints will be described below.
 
-TBD
+Events are only scheduled in slots they are available for
```
Comment (meatballs) [inline docs/background/mathematical_model.rst]: grammar: 'they are available for' should be 'for which they are available'

File: docs/background/mathematical_model.rst [hunk 3]
```diff
@@ -55,4 +59,168 @@ be a bit more complicated:
...
+
+Talks in a given session have something in common
+-------------------------------------------------
+
+It might be desirable to schedule collection of time slots in such a way that
+the events in that collection have something in common. Perhaps all talks in a
+morning session in a particular room should be welcoming to delegates of a given
+level of expertise.
+
+To do this we first need to capture each collection of slots in to sessions, and
```
Comment (meatballs) [inline docs/background/mathematical_model.rst]: typo: 'in to' should be 'into'

File: docs/background/mathematical_model.rst [hunk 4]
```diff
@@ -55,4 +59,168 @@ be a bit more complicated:
...
++++++++++++++++++++
+
+Optimising to avoid room overflow
+---------------------------------
+
+Demand for events might be known: this will be captured using a vector
+:math:`d\in\mathbb{R}_{\geq 0}^{M}`. Similarly capacity for rooms might be
+known, captured using another vector :math:`c\in\mathbb{R}_{\geq 0}^{N}`. Whilst
+it might not be possible to stick to those constraints strongly (when dealing
+with parallel sessions delegate might not go where they originally intended) we
```
Comment (meatballs) [inline docs/background/mathematical_model.rst]: typo: delegates

File: docs/background/mathematical_model.rst [hunk 5]
```diff
@@ -55,4 +59,168 @@ be a bit more complicated:
...
+can aim to minimise the expected overflow given by the following expression:
+
+.. math::
+   :label: overflow_objective_function
+
+   \sum_{i=1}^{M}\sum_{j=1}^{N}X_{ij}(c_j - d_i)
+
+Using this our optimisation problem to give a desirable schedule is obtained by
```
Comment (meatballs) [inline docs/background/mathematical_model.rst]: typo missing comma: 'Using this, our....'

File: docs/background/mathematical_model.rst [hunk 6]
```diff
@@ -55,4 +59,168 @@ be a bit more complicated:
...
+solving the following problem:
+
+Minimise :eq:`overflow_objective_function` subject to :eq:`all_events_scheduled_constraint`,
+:eq:`all_slots_at_most_1_event_constraint`,
+:eq:`slot_constraint`, :eq:`event_constraint` and :eq:`tag_constraint`.
+
+Minimise change from a previous schedule
+----------------------------------------
+
+Once a schedule has been obtained and publicised to all delegates a new
```
Comment (meatballs) [inline docs/background/mathematical_model.rst]: typo: missing comma '...publicised to all delegates, a new...'

File: docs/background/mathematical_model.rst [hunk 7]
```diff
@@ -55,4 +59,168 @@ be a bit more complicated:
...
+constraint might arise (modifying :eq:`all_events_scheduled_constraint`,
+:eq:`all_slots_at_most_1_event_constraint`,
+:eq:`slot_constraint`, :eq:`event_constraint` and :eq:`tag_constraint`). At this
+point the original optimisation problem can be solved again leading to a
+potentially completely different schedule. An alternative to this is to use
+distance from an original schedule :math:`X_o` as the objective function. Norms
+on matrix spaces are usually non linear however given the boolean nature of our
+variables the following function can be used to measure the number of changes:
```
Comment (meatballs) [inline docs/background/mathematical_model.rst]: missing commas: 'however, given the boolean nature of our variables, the following function'

File: docs/background/mathematical_model.rst [hunk 8]
```diff
@@ -55,4 +59,168 @@ be a bit more complicated:
...
+
+   \delta(x_o, x) =
+   \begin{cases}
+       x,&\text{ if } x_o=0\\
+       1-x,&\text{ if } x_o=1
+   \end{cases}
+
+
+Using this it is possible to obtain a schedule that is least disruptive from
+another schedule when presented with new constrains by solving the following
```
Comment (meatballs) [inline docs/background/mathematical_model.rst]: typo: constraints

File: docs/background/mathematical_model.rst (hunk 3 above)
Comment (drvinceknight) [inline docs/background/mathematical_model.rst]: I swear they change this one all the time ;)

**Pull request merged**